core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -4195,34 +4194,44 @@ public long getGlobalCheckpoint() {
         indexSettings.updateIndexMetaData(builder.build());
 
         store = createStore();
-        try (InternalEngine engine
-                 = createEngine(indexSettings, store, createTempDir(), NoMergePolicy.INSTANCE, null, seqNoServiceSupplier)) {
-            int numDocs = scaledRandomIntBetween(10, 100);
-            for (int docId = 0; docId < numDocs; docId++) {
-                ParseContext.Document document = testDocumentWithTextField();
-                document.add(new Field(SourceFieldMapper.NAME, BytesReference.toBytes(B_1), SourceFieldMapper.Defaults.FIELD_TYPE));
-                engine.index(indexForDoc(testParsedDocument(Integer.toString(docId), null, document, B_1, null)));
-                if (frequently()) {
-                    globalCheckpoint.set(randomIntBetween(
-                        Math.toIntExact(engine.seqNoService().getGlobalCheckpoint()),
-                        Math.toIntExact(engine.seqNoService().getLocalCheckpoint())));
+        engine = new InternalEngine(config(indexSettings, store, createTempDir(), NoMergePolicy.INSTANCE, null), seqNoServiceSupplier) {
+            @Override
+            protected void commitIndexWriter(IndexWriter writer, Translog translog, String syncId) throws IOException {
+                // The global checkpoint is advanced but not fsynced yet.
+                final long lagging = seqNoService().getLocalCheckpoint() - seqNoService().getGlobalCheckpoint();
```

Comment:
I don't follow this - why do we need to query the global checkpoint from the seqNoService? we alread have access to globalCheckpoint ?

I think what you mean here is something like this - no if no rarely:

```
 // Advance the global checkpoint during the flush to create a lag between what's persisted in the translog (and is visible for CombinedDeletionPolicy) and what's in memory in the SequenceServices
globalCheckpoint.set(randomLongBetween(globalCheckpoint.get(), seqNoService().getLocalCheckpoint()));
```

wdyt?